File: RandomDataPicker/Models/Entry.cs
namespace RandomDataPicker.Models;

[MessagePack.MessagePackObject(true)]
public record Entry
{
    public int? Id { get; set; }
    public string? Name { get; set; }
    public string? Email { get; set; }
    public string? City { get; set; }
    public bool IsFlagged { get; set; }
    public bool LogicallyEquals(Entry entry)
    {
        return entry != null &&
            !string.IsNullOrWhiteSpace(entry.Name) && entry.Name.Equals(Name) &&
            !string.IsNullOrWhiteSpace(entry.Email) && entry.Email.Equals(Email)
            && !string.IsNullOrWhiteSpace(entry.City) && entry.City.Equals(City);
    }
}

File: App/src/models/entry.ts
export interface IEntry {
    id:number;
    name?:string;
    email?:string;
    city?:string;
    isFlagged:boolean;
}

export class Entry implements IEntry {
    id:number;
    name?:string;
    email?:string;
    city?:string;
    isFlagged:boolean;
    
    constructor() {
        this.id = 0;
        this.isFlagged = false;
    }

    static convert(entry:IEntry): Entry {
        const newEntry = new Entry();
        newEntry.city = entry.city;
        newEntry.email = entry.email;
        newEntry.id = entry.id;
        newEntry.isFlagged = entry.isFlagged;
        newEntry.name = entry.name;
        return newEntry;
    }
}